src/Roots.jl

```diff
@@ -13,30 +16,312 @@ import ClimaLSM:
     auxiliary_vars,
     initialize,
     initialize_auxiliary
-export RootsModel
+export RootsModel,
+    compute_flow,
+    theta_to_p,
+    p_to_theta,
+    RootsPrescribedExchange,
+    RootsParameters
+
+"""
+    RootsParameters{ft <: AbstractFloat}
+
+
+A struct for holding parameters of the Root Model. Eventually to be used with ClimaParameters.
+"""
+struct RootsParameters{ft <: AbstractFloat}
+    a_root::ft
+    b_root::ft
+    a_stem::ft
+    b_stem::ft
+    size_reservoir_stem_moles::ft
+    size_reservoir_leaf_moles::ft
+    K_max_root_moles::ft
+    K_max_stem_moles::ft
+end
+
 
 """
     RootsModel
 
-A standin for a model used to simulate water content in roots.
+Defines, and constructs instances of, the RootsModel type, which is used
+for simulation flow of water to/from soil, along roots of different depths,
+along a stem, to a leaf, and ultimately being lost from the system by
+transpiration. 
+
+This model can be used in standalone mode by prescribing the transpiration rate
+and soil pressure at the root tips (boundary_exchanges of type `RootsPrescribedExchange`),
+or with a dynamic soil model (boundary exchanges of type TBD).
+
+The RootModel domain must be of type `AbstractPlantDomain`.
 """
-struct RootsModel{FT, ps, d} <: AbstractModel{FT}
+struct RootsModel{FT, ps, d, b} <: AbstractModel{FT}
     param_set::ps
     domain::d
+    boundary_exchanges::b
     model_name::Symbol
 end
 
 function RootsModel{FT}(;
     param_set = param_set,
-    domain = domain <: AbstractDomain{FT},
+    domain::AbstractPlantDomain{FT} = domain,
+    boundary_exchanges::AbstractComponentExchange{FT} = boundary_exchanges,
```

Comment:
Another small style thing - for each keyword argument it's not necessary to have, e.g. `param_set = param_set`, we can just have `param_set` as a keyword and there is no default.